hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/blockmanagement/BlockManager.java

```diff
@@ -725,6 +743,9 @@ public void activate(Configuration conf, long blockTotal) {
     datanodeManager.activate(conf);
     this.redundancyThread.setName("RedundancyMonitor");
     this.redundancyThread.start();
+    this.markedDeleteBlockScrubberThread.
+        setName("MarkedDeleteBlockScrubberThread");
+    this.markedDeleteBlockScrubberThread.start();
```

Comment:
Hi @zhuxiangyi , I have a question.
I noticed that the markedDeleteBlockScrubberThread has started, but I didn't see it execute close() anywhere. Because there are some similar worker threads in BlockManager: redundancyThread. They are all reflected in BlockManager#close().
Is there a difference between markedDeleteBlockScrubberThread?

Reply:
@jianghuazhu  Thanks a lot for your review.
The correct BlockManager#close() needs to interrupt the markedDeleteBlockScrubberThread thread.

Reply:
Thanks @zhuxiangyi  for the comment.
It seems that markedDeleteBlockScrubberThread will run in active nn and standby nn at the same time. Is that right?

Reply:
Correct, both active nn and standby will run it.